engine/src/main/java/org/hibernate/validator/internal/engine/tracking/PredefinedScopeProcessedBeansTrackingStrategy.java

```diff
@@ -160,38 +159,38 @@ private boolean determineTrackingRequired(Class<?> beanClass, Set<Class<?>> bean
 
 		// TODO: is there a more concise way to do this?
 		private <T> Set<Class<?>> getDirectCascadedBeanClasses(Class<T> beanClass ) {
-			final BeanMetaData<T> beanMetaData = beanMetaDataManager.getBeanMetaData( beanClass );
-			if ( beanMetaData.hasCascadables() ) {
-				final Set<Class<?>> directCascadedBeanClasses = new HashSet<>();
-				for ( Cascadable cascadable : beanMetaData.getCascadables() ) {
-					final CascadingMetaData cascadingMetaData = cascadable.getCascadingMetaData();
-					if ( cascadingMetaData.isContainer() ) {
-						final ContainerCascadingMetaData containerCascadingMetaData = (ContainerCascadingMetaData) cascadingMetaData;
-						if ( containerCascadingMetaData.getEnclosingType() instanceof ParameterizedType ) {
-							ParameterizedType parameterizedType = (ParameterizedType) containerCascadingMetaData.getEnclosingType();
-							for ( Type typeArgument :  parameterizedType.getActualTypeArguments() ) {
-								if ( typeArgument instanceof Class ) {
-									directCascadedBeanClasses.add( (Class<?>) typeArgument );
-								}
-								else {
-									throw new UnsupportedOperationException( "Only ParameterizedType values of type Class are supported" );
-								}
+			final BeanMetaData<?> beanMetaData = rawBeanMetaDataMap.get( beanClass );
+
+			if ( beanMetaData == null || !beanMetaData.hasCascadables() ) {
```

Comment:
I was wondering how to deal with `beanMetaData == null`. I'm glad to see that spelled out here.